spec/ruby/core/module/autoload_spec.rb

```diff
@@ -11,6 +11,18 @@
   it "returns nil if no file has been registered for a constant" do
     ModuleSpecs::Autoload.autoload?(:Manualload).should be_nil
   end
+
+  it "returns the name of the file that will be autoloaded if an ancestor defined that autoload" do
+    ModuleSpecs::Autoload::Parent.autoload :AnotherAutoload, "another_autoload.rb"
+    ModuleSpecs::Autoload::Child.autoload?(:AnotherAutoload).should == "another_autoload.rb"
+  end
+
+  ruby_version_is "2.7" do
+    it "returns nil if an ancestor defined that autoload but recursion is disabled" do
+      ModuleSpecs::Autoload::Parent.autoload :AnotherAutoload, "another_autoload.rb"
+      ModuleSpecs::Autoload::Child.autoload?(:AnotherAutoload, false).should be_nil
+    end
```

Comment:
It would be useful to also test inherit=false with a case that doesn't return `nil`, to make the sure the logic is correct too in that case. Could you add a spec for it?

Reply:
Of course.

Reply:
Done.